### DISCUSSION TITLE
react-hook-form: how to remove the item (which is mentioned in defautvalue) from submit data if its not mounted. Tried unregister its not working

### DISCUSSION THREAD

**Discussion by sant527** [react-hook-form/react-hook-form#7146] [Q&A] [answered]
I have simple form with `firstName` and `lastName`. I have added some `defaultValues` to them in `useForm`. I havent mounted the `lastName` using if condition. 

Now when I try to submit, I am expecting it to show only the mounted components values i.e `firstName`. But it shows both `firstName` and `lastName`. I have created a button to try to `unregister("lastName")` but even after unregister and not being mounted it shows in the submit data.

If I dont provide `defaultValues` then it works well, like if not mounted initially it will not be shown in the submit data.

Below is the code

```
import React, { useState } from "react";
import ReactDOM from "react-dom";
import { useForm } from "react-hook-form";

import "./styles.css";

const App = () => {
  const { register, handleSubmit, unregister, watch } = useForm({
    defaultValues: {
      firstName: "test",
      lastName: "test2"
    }
  });

  const onSubmit = (data) => {
    console.log("ON SUBMIT");
    console.log(JSON.stringify(data, null, 4));
  };


  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>Last Name</label>
      <input {...register("firstName")} />
      {false && (
        <>
          <label>Last Name</label>
          <input {...register("lastName")} />
        </>
      )}
      <button
        type="button"
        onClick={() => {
          console.log("unregistering lastName")
          unregister("lastName");
        }}
      >
        unregister lastName
      </button>
      <input type="submit" />
    </form>
  );
};

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);
```

And the [codesandbox][1]

[![enter image description here][2]][2]

  [1]: https://codesandbox.io/s/react-hook-form-unregister-v6-forked-rriop?file=/src/index.js:0-1038
  [2]: https://i.stack.imgur.com/LmMNK.png

**Comment by ritikbanger** [comment, marked as answer]
Make sure you unmount that input which has register callback or else the input will get registered again. Due to default values, the lastname is registered again every time and you get it as output.

**Reply by bluebill1049** [reply]
That's exactly right. I have a video tutorial coming up for this as well. if your input doesn't gets unmounted, the register function will get invoked again.
